### PR TITLE
dev/core#1172 Fix Contribution import to handle ContributionSoft in a storable way

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
dev/core#1172 Fix Contribution import to handle ContributionSoft in a storable way

Before
----------------------------------------
When importing with a saved mapping it is not possible to store the soft_credit_type _id

After
----------------------------------------
The soft_credit_type_id is stored to the mapping

Technical Details
----------------------------------------
There is a bit going on here. We discussed adding `entity_metadata` to the `civicrm_mapping_field` table to store this additional info. That PR is still open & I was on the fence & decided this is a good approach for now - instead of storing

field_name `soft_credit`
contact_type `external_identifier`
not saved `soft_credit_type_id`

it now saves 

`field_name` `soft_credt.contact.external_identifer`
`contact_type` `{'soft_credit' => {'soft_credit_type_id' => 9}}`

In other words everything we know about the field is now in the `field_name` and the rest is in the `contact_type`. I kinda wanted to drop contact type for `entity_type` & I think that would be OK - but I also concluded that medium term we might be able to move this stuff to the user_job table with `is_template` set to true. So this is slightly less butchery & doesn't lose the data.

In combination with previous work & added tests this allows the `import` function to be significantly stripped down

Comments
----------------------------------------

### CODE REVIEW

**Comment by colemanw** [comment]
Ok, I don't want to nitpick because @eileenmcnaughton is the one putting in all the work.

**Comment by eileenmcnaughton** [comment]
thanks - yeah there are a couple of things that could change again a bit - it's incremental
